aten/src/ATen/native/Sorting.cpp

```diff
@@ -182,13 +183,23 @@ void quantile_impl(
   TORCH_CHECK(
       sorted.size(-1) <= std::pow(2, 24),
       "quantile() input tensor is too large");
-  
-  Tensor ranks = q * (sorted.size(-1) - 1);
+
+  Tensor ranks;
+  if (ignore_nan) {
+    ranks = q * (sorted.isnan().logical_not().sum(-1, true) - 1);
```

Comment:
Add a comment explaining what's up here. I think you can do 

`sorted.isnan().logical_not_()...` to avoid one of these tensors from being created.

Reply:
It's too bad this has to use `isnan().logical_not_()`. I would think that `torch.isreal()` would do something like that, but NumPy's `isreal`, which ours is based on, says NaNs are real:

```
import numpy as np
np.isreal(float('nan'))
: True
```

Which seems wrong, but oh well.